### PR TITLE
Chloe development

### PR DESCRIPTION
**Pull request by chloemiranz**
Working comments with datastore.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -15,18 +15,77 @@
...
+    private final ArrayList<String> comments = new ArrayList<String>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+   
+    listData.add("Hello!"); 
+    listData.add("How are you?");
+    listData.add("Great weather today!");
+
+    //converting data to JSON
```
Comment (Ali09) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: "Converting data to JSON."

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -26,3 +27,55 @@ function addRandomGreeting() {
...
+    const dataElements = document.getElementById('data-container');
+    statsListElement.innerHTML = '';
+    statsListElement.appendChild(
+        createListElement('Element 1: ' + data.get(0)));
+    statsListElement.appendChild(
+        createListElement('Element 2: ' + data.get(1)));
+    statsListElement.appendChild(
+        createListElement('Element 3: ' + data.get(2)));
+  });
+}*/
```
Comment (Ali09) [inline portfolio/src/main/webapp/script.js]: Nit: Probably want to clean up commented out functions and lines, e.g. here and a few lines your doPost function.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -15,18 +15,77 @@
...
+  }
+
+  private String convertToJson(ArrayList<String> data) {
+    String json = "";
+    json += data.get(0);
+    json += ", ";
+    json += data.get(1);
+    json += ", ";
+    json += data.get(2);
+    json += "";
```
Comment (Ali09) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: You could probably generalize this with a for-loop to work for ArrayLists of any size using less lines of code.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -15,18 +15,68 @@
...
+
+    String json = convertToJson(listData);
+    
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+
+  }
+
+  private String convertToJson(ArrayList<String> data) {
```
Comment (MrM117) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: FYI: there's also the Gson library you could use: https://github.com/google/gson/blob/master/UserGuide.md

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -26,3 +27,39 @@ function addRandomGreeting() {
...
+  var imageContainer = document.getElementById('random-image-container');
+  // Remove the previous image.
+  imageContainer.innerHTML = '';
+  imageContainer.appendChild(imgElement);
+}
+
+function showFrame() {
+    document.getElementById("showiFrame").innerHTML = "<iframe src=\"https://open.spotify.com/embed/track/1bSpwPhAxZwlR2enJJsv7U\" width=\"300\" height=\"380\" frameborder=\"0\" allowtransparency=\"true\" allow=\"encrypted-media\" ></iframe>";
+
+
```
Comment (MrM117) [inline portfolio/src/main/webapp/script.js:50]: nit: delete blank lines

**Review by MrM117 — APPROVED** [review]

**Pull request merged**